src/server/engine_shard.cc

```diff
@@ -735,11 +735,14 @@ void EngineShard::Heartbeat() {
   DbSlice& db_slice = namespaces->GetDefaultNamespace().GetDbSlice(shard_id());
   // Skip heartbeat if we are serializing a big value
   static auto start = std::chrono::system_clock::now();
-  if (db_slice.WillBlockOnJournalWrite()) {
+  // We acquire exlusive locks during global transactions
+  // If we can't acquire this it means the shard is under global lock.
+  const bool can_acquire_global_lock = shard_lock()->Check(IntentLock::Mode::EXCLUSIVE);
```

Comment:
note, we don't acquire here, namely we don't increment `Cx` and `Cs` (exlusive and shared intent locks). We merely test them to understand if we are under global lock or not.